File: sdks/thunkmetrc-cs/ThunkMetrc.Wrapper/Models/CreateAdditivesTemplatesRequest.cs
using System;
using System.Collections.Generic;

namespace ThunkMetrc.Wrapper.Models
{
    public class CreateAdditivesTemplatesRequest
    {
        [global::System.Text.Json.Serialization.JsonPropertyName("ActiveIngredients")]
        public List<object>? ActiveIngredients { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("AdditiveType")]
        public string? AdditiveType { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ApplicationDevice")]
        public string? ApplicationDevice { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("EpaRegistrationNumber")]
        public string? EpaRegistrationNumber { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("Name")]
        public string? Name { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("Note")]
        public string? Note { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ProductSupplier")]
        public string? ProductSupplier { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ProductTradeName")]
        public string? ProductTradeName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RestrictiveEntryIntervalQuantityDescription")]
        public string? RestrictiveEntryIntervalQuantityDescription { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RestrictiveEntryIntervalTimeDescription")]
        public string? RestrictiveEntryIntervalTimeDescription { get; set; }
        public class CreateAdditivesTemplatesRequestActiveIngredientsItem
        {
            [global::System.Text.Json.Serialization.JsonPropertyName("Name")]
            public string? Name { get; set; }
            [global::System.Text.Json.Serialization.JsonPropertyName("Percentage")]
            public double? Percentage { get; set; }
        }
    }
}

File: sdks/thunkmetrc-kotlin/wrapper/src/main/kotlin/io/github/thunkier/thunkmetrc/wrapper/models/CreateAdditivesTemplatesRequest.kt
package io.github.thunkier.thunkmetrc.wrapper.models

import com.fasterxml.jackson.annotation.JsonProperty



data class CreateAdditivesTemplatesRequest(
    @JsonProperty("ActiveIngredients")
    val activeIngredients: List<Any>? = null,
    @JsonProperty("AdditiveType")
    val additiveType: String? = null,
    @JsonProperty("ApplicationDevice")
    val applicationDevice: String? = null,
    @JsonProperty("EpaRegistrationNumber")
    val epaRegistrationNumber: String? = null,
    @JsonProperty("Name")
    val name: String? = null,
    @JsonProperty("Note")
    val note: String? = null,
    @JsonProperty("ProductSupplier")
    val productSupplier: String? = null,
    @JsonProperty("ProductTradeName")
    val productTradeName: String? = null,
    @JsonProperty("RestrictiveEntryIntervalQuantityDescription")
    val restrictiveEntryIntervalQuantityDescription: String? = null,
    @JsonProperty("RestrictiveEntryIntervalTimeDescription")
    val restrictiveEntryIntervalTimeDescription: String? = null
)
